### DISCUSSION TITLE
editable `DataGridTemplateColumn`

### DISCUSSION THREAD

**Discussion by timunie** [AvaloniaUI/Avalonia#6553] [Q&A] [answered]
Hello Avalonia Team, 

I am looking for a way to create a `DataGridTemplateColumn` which supports a different view for the editing mode. In WPF there are two different `DataTemplates` for the normal view (`CellTemplate`) and the editing view (`CellEditingTemplate`). See: https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.datagridtemplatecolumn?view=net-5.0 . 

My questions are: 
- What would be the way in Avalonia if there is any?
- If there is none, is that a feature you also find useful? I can open a feature request then.

Thank you and happy coding
Tim

**Comment by maxkatz6** [comment, marked as answer]
In Silverlight/UWP and in its port to Avalonia in this repo there is no "edit" state for the template column. You basically put any content into the template what you want. And 'edit' state can be defined by control being focused, for example. 
Still, it might be useful to have separated template for edit state, and at first support edit state for template column at all. But it might be a bit complex work and also will be a breaking change (unless CellEditingTemplate will always be the same as CellTemplate by default). 

Not sure if we need it, especially when developer can workaround it having full control over the template.

**Reply by timunie** [reply]
Hi @maxkatz6 

Thank you for the detailed answer and your ideas to work around this. I for example have a column showing a date as a TextBlock. I want the user to get a DatePicker after double click just like inbthe other columns. I will be able to do what I want to do, so if you are not interested in this feature I am fine with it.    But I want to share my idea how to add this function without a break change:

I would implement a `DataGridEditableTemplateColumn` which is based on `DataGridTemplateColumn`. 

Happy coding
Tim

**Comment by timunie** [comment]
Hi @maxkatz6 

I have implemented the feature and tested it in a private App and everything seems to work quite well. I extended the `DataGridTemplateColumn` to get what I was looking for. If you find this may be useful to any other user, I will provide a PR (so you can test the solution and finally decide). The benefits for me are that I get the same behavior as any other cell has, so double-click and F2 will enter the editing state. 

What do you or anyone else think? If I should provide a PR just leave a 👍 as reaction. 

Happy coding 
Tim

**Reply by maxkatz6** [reply]
@timunie yes, I can take a look. Also, I would prefer creating CellEditingTemplate column with null by default, so it won't need to change any templates in this case (so no breaking changes). With these conditions it can be merged easily, if works well.
